Jk unit tests

Bsubs opened the pull request:
1. Cleaned up queries in /user route
2. Added tests for User, Scorelist and Scorecard database models

Bsubs committed:
Updated queries in /user route

Bsubs committed:
Added test for User model

Bsubs committed:
Added test for scorelists

Bsubs committed:
Cleaned up queries in user route

Bsubs committed:
Added tests for scorecards

Bsubs committed:
Fixed changes as indicated by comments

jwyatt1999 commented on routes.py:
```diff
@@ -40,21 +40,11 @@ def credits():
-    scores_freeplay = Scorecard.query.filter(Scorecard.type == 2).filter(Scorecard.scorelist_id == user.scorelists.id).order_by(Scorecard.date.desc())
-    return render_template('profile_page.html', user=user, score_freeplay = scores_freeplay, score_daily = scores_daily, total_avg=total_avg, total_games=total_games, 
-    freeplay_avg=freeplay_avg, freeplay_games=freeplay_games, daily_avg=daily_avg, daily_games=daily_games, freeplay_highscore=freeplay_highscore, daily_highscore=daily_highscore)
+    total_avg = round(Scorecard.query.with_entities(func.avg(Scorecard.score)).filter(Scorecard.uname == user.username).all()[0][0])
```
I think it would be fine to have 2 decimal places instead of 0. It gives the user a better idea of their average without being "too much" information

jwyatt1999 commented on templates/profile_page.html:
```diff
@@ -24,18 +24,18 @@ <h1>Welcome back {{user.username}}!</h1>
             <h3>The highest score you achieved in this mode:</h3>
             <ul>
-                <li class="list-group-item">{{ daily_highscore }}</li>
+                <li class="list-group-item">{{ user.daily_scorecards().first() }}</li>
```
I expect this will give the wrong result. The array is sorted by date, so first will be the most recent score, not the highest. Have you tested this with a high score game and then a lower scored game?

jwyatt1999 commented on templates/profile_page.html:
```diff
@@ -44,18 +44,18 @@ <h3>Your Recent Daily Scores:</h3>
             <h3>The highest score you achieved in this mode:</h3>
             <ul>
-                <li class="list-group-item">{{ freeplay_highscore }}</li>
+                <li class="list-group-item">{{ user.freeplay_scorecards().first() }}</li>
```
Similarly to above, I don't expect this to give the user's highest scoring freeplay game

jwyatt1999 commented on test.py:
```diff
@@ -0,0 +1,90 @@
+        db.drop_all()
+
+    def test_password_hashing(self):
+        u = User(username='JooKai', email='joo@gmail.com')
```
I'm thinking we should replace the emails and usernames in this file with 'test1@gmail.com' and so on. Just makes the tests a bit easier to read / understand in my opinion

jwyatt1999 commented:
Looks pretty good overall, just found a few things that I think should be changed. Have you tested running unittester?

Bsubs commented:
I've fixed the code according to the comments:
1. Changed rounding to 2 decimal places
2. Fixed methods to return correct highscore
3. Changed values in unittester and added new tests for the methods introduced in point 2

jwyatt1999 commented:
Nice work!
